packages/sdk/src/rtl/browserTransport.ts

```diff
@@ -192,6 +79,106 @@ export class BrowserTransport extends BaseTransport {
     super(options)
   }
 
+  /** Does this browser have the necessary performance APIs? */
+  static supportsPerformance() {
+    return (
+      performance &&
+      performance.mark !== undefined &&
+      performance.measure !== undefined
+    )
+  }
+
+  private static _trackPerf = false
+
+  public static get trackPerformance() {
+    return this._trackPerf
+  }
+
+  public static set trackPerformance(value: boolean) {
+    this._trackPerf = value && BrowserTransport.supportsPerformance()
+  }
+
+  static startMark = 'A'
+  static endMark = 'B'
+
+  /**
+   * Create a performance mark
+   * @param name to use as prefix of mark. Use `markName()` to determine the name
+   * @param tag tag to use to distinguish mark
+   */
+  static mark(name: string, tag: string) {
+    if (this.trackPerformance) {
+      const mark = `${name}-${tag}`
+      performance.mark(mark)
+      console.debug(`Marked ${mark}`)
+    }
+  }
+
+  /**
+   * Create a unique name for the performance metric of the url based on its start time
+   * @param url metric to find
+   *
+   * Uses the last entry of the matching metric to create the name for the tracker so
+   * post processing can use it to merge into the main resource tracker
+   *
+   */
+  static markName(url: string) {
+    if (!this._trackPerf) return url
+
+    const entries = performance.getEntriesByName(url, 'resource')
+    if (entries.length > 0) {
+      const last = entries[entries.length - 1]
+      return `${url}-${last.startTime}`
+    }
+    return url
+  }
```

Comment:
This is the crux of accurate performance tracking issues: sometimes the resource timing performance entry for the URL is not complete at the time `markName()` is called, so it defaults to the more ambiguous _url as name_ pattern rather than including the start time used to uniquely identify duplicate URL requests.

Also, in `endMark()` sometimes there is a second resource timer that exists for the same url before `endMark()` is called, which leads to `markName()` finding the second one and losing the `startTime` tag.

This leads to `endMark()` not being able to associate with the correctly produced start mark to create the measure for the correct request.

There is some latency in the performance collections as well for retrieval and deleting.

So, I'm looking for an unambiguous, thread-safe way to track the performance metric. Maybe adding `startMark? :string` to `IRawResponse` and using that in `markEnd()` will work. I don't think https://developer.mozilla.org/en-US/docs/Web/API/Performance/now provides the level of timing precision we need since it rounds to the millisecond. I'll just need a thread-safe `startMark` tag creator.